<launch>
    <!-- 启动 USB 相机 -->
    <node name="usb_cam" pkg="usb_cam" type="usb_cam_node" launch-prefix="bash -c 'rosrun usb_cam usb_cam_node _video_device:=/dev/video0 _pixel_format:=mjpeg _image_width:=1280 _image_height:=480 2>/dev/null; $0 $@'" />

    <!-- 图像左右分割节点 -->
    <node pkg="stereo_splitter" type="split_image_info.py" name="split_stereo_image" output="screen" />
        <param name="left_camera_yaml" value="/home/xhy/catkin_ws/camera/left.yaml"/>
        <param name="right_camera_yaml" value="/home/xhy/catkin_ws/camera/right.yaml"/>
        <param name="input_image_topic" value="/usb_cam/image_raw"/>

    <!-- 左目图像去畸变 -->
    <node pkg="image_proc" type="image_proc" name="left_image_proc" output="screen">
        <remap from="image_raw" to="/left/image_raw"/>
        <remap from="camera_info" to="/left/camera_info"/>
    </node>

    <!-- 右目图像去畸变 -->
    <node pkg="image_proc" type="image_proc" name="right_image_proc" output="screen">
        <remap from="image_raw" to="/right/image_raw"/>
        <remap from="camera_info" to="/right/camera_info"/>
    </node>

    <!-- stereo_image_proc 用于立体匹配，生成 disparity 和 point cloud -->
    <node pkg="stereo_image_proc" type="stereo_image_proc" name="stereo_proc" output="screen">
        <!-- 使用 image_proc 去畸变之后的话题作为输入 -->
        <remap from="left/image_rect" to="/left/image_raw/image_rect_color"/>
        <remap from="right/image_rect" to="/right/image_raw/image_rect_color"/>
        <remap from="left/camera_info" to="/left/camera_info"/>
        <remap from="right/camera_info" to="/right/camera_info"/>
    </node>


</launch>

